### ISSUE TITLE
Using `C++2a` as a `cppdialect` triggers the Default dialect (C++14)

### ISSUE THREAD

**Issue by lucho1** [premake/premake-core#2030]
**What seems to be the problem?**
I wanted to use C++20 Draft version as C++ dialect in my project (the latest C++ available), and I specified `C++2a` as the `cppdialect` for the project, following [this](https://premake.github.io/docs/cppdialect) documentation that states it's the value to use for that.

However, once I generated the project in Visual Studio 2019, I found that the language is the default `Default (ISO C++14 Standard)`. This might be the default version triggered, so I initially thought it was something I was doing wrong, but just changing the `cppdialect` to `C++latest` actually works and enables the C++20 Draft.

I also checked the [GitHub documentation](https://github.com/premake/premake-core/wiki/cppdialect) for this to see if there was something wrong, and apparently it looks like it's still under construction.

I made a screenshot to illustrate the problem:
![image](https://user-images.githubusercontent.com/11302677/215733890-bcf3e70e-f8e3-422b-97d2-5e0ac5f28268.png)
![image](https://user-images.githubusercontent.com/11302677/215742574-9ee426d3-7a18-414b-ad14-318ae6bbcaa8.png)

The current configuration I have in the premake is the next one:

```
workspace "WksName"
	... [some specifications here]...

project "ProjectName"
	kind "ConsoleApp"
	language "C++"
	cppdialect "C++2a"
	staticruntime "on"
	... [more specifications here]...
```

The `...[more specifications here]...` is just stuff that I'm configuring and removing here for clarity (they work properly always). Using the `latest` label will work fine for me now, but I think this might still be worth flagging.

**What did you expect to happen?**
`C++2a` triggers usage of C++20 Draft.

**What have you tried so far?**
Using `C++2a` triggers `Deafult (C++14)`, using `C++latest` triggers C++20 Draft (actually shown as `Preview - Features from the Latest C++ Working Draft (/std:c++latest)`.

**How can we reproduce this?**
By specifying `C++2a` as a `cppdialect` and generating the project (for me it's Visual Studio 2019, not sure about other IDEs or versions), then checking the C++ version used.

- [x] Visual Studio 2019 (vs2019)

**What version of Premake are you using?**
5.0.0-beta2

**Comment by nickclark2016** [comment]
Where are you seeing that `elseif` at @Jarod42?  Current language standard related code in VS exporter doesn't look like that.

**Comment by lucho1** [comment]
Hi @Jarod42 & @nickclark2016 !
Thanks for the quick answers.

Yeah, I wanted to use C++20 initially, but then I had to switch to the latest C++20 Draft to use one part of the standard library that's only there. As I say, using `C++latest` works for that, so it's fine, but I just wanted to flag this in case there was something to do about it.

**Comment by nickclark2016** [comment]
I think the initial thought on the `cppdialect` flag was that we'd only support the language versions explicitly called out by the toolchain and `C++latest`, so we didn't support the GNU and C++ draft versions in VS.

**Comment by Jarod42** [comment]
@nickclark2016: It is there [modules/vstudio/vs2010_vcxproj.lua#L1634](https://github.com/premake/premake-core/blob/master/modules/vstudio/vs2010_vcxproj.lua#L1634).

Ideally, I think exporters should warn about unsupported fields. (pretty sure we will disagree on that ;-) )

At minima, documentation should be updated.

**Comment by nickclark2016** [comment]
Nah, I agree that they should warn. We just don't have the framework in place currently to do that.

**Comment by nickclark2016** [comment]
> @nickclark2016: It is there [modules/vstudio/vs2010_vcxproj.lua#L1634](https://github.com/premake/premake-core/blob/master/modules/vstudio/vs2010_vcxproj.lua#L1634).
> 
> Ideally, I think exporters should warn about unsupported fields. (pretty sure we will disagree on that ;-) )
> 
> At minima, documentation should be updated.

Ahh, I see what you're referring to now. Yeah, that doesn't really affect VS project files for 2019 and 2022, so the behavior appears correct to me. We use `m.languageStandard` to set the language standard in VS 2017 and newer.

**Comment by Jarod42** [comment]
Indeed, I miss `m.languageStandard`, so just doc to update, and warning later when supported. (A simple print would seem enough for me).

**Comment by nickclark2016** [comment]
To confirm, doc to update == website, correct?

**Comment by Jarod42** [comment]
yes.

**Comment by lucho1** [comment]
Hi again, thanks for taking care of this so quick, and sorry for my late responses, I don't check GitHub as often as I'd like 😅

So if I understood correctly, there would be no support for draft versions in Premake? Not sure if I understood it properly as I get a bit lost in the messages since I'm new to all this :)

**Comment by nickclark2016** [comment]
Visual Studio does not support draft versions.  If you want C++20, just use that.  If you want the latest, use C++Latest.

**Comment by lucho1** [comment]
Okay, so when you're selecting `Features from the Latest C++ Working Draft (std:c++latest)` in the VS properties it doesn't use a draft version? Or I guess it's just using only the latest draft but doesn't lets you chose a draft version?

Just asking as I'm not an expert in this, I tried checking [VS Documentation](https://learn.microsoft.com/en-gb/cpp/build/reference/std-specify-language-standard-version?view=msvc-170#c-standards-support) but still I don't know how to relate it to your answer.

**Comment by nickclark2016** [comment]
Correct, VS ignores draft revisions.  If you want the latest draft, you use `C++Latest`, not `C++2a`.  However, Visual Studio 2022 supports C++20 completely, so I'd recommend using `C++20` instead of `C++Latest`, as the latest draft implementation includes things that are coming in C++23 (C++2b).

**Comment by lucho1** [comment]
Oh okay, thanks for that information!

I was actually using `C++20`, but I wanted to use `<format>` and `<chrono>` libraries that I think are available only with the latest draft `C++Latest` and not with the standard `C++20` version, so that's why I switched to the latest draft

**Comment by nickclark2016** [comment]
Those are headers in C++20.

**Comment by lucho1** [comment]
No, you can't include them with the standard version of C++20 (it actually says it in the [VS docs](https://learn.microsoft.com/en-gb/cpp/build/reference/std-specify-language-standard-version?view=msvc-170#c-standards-support)), and not being able to include them was why I changed. The chrono one it is, but not its formatting side I think, which is under C++Latest.

**Comment by Jarod42** [comment]
There are still some [MACROs](https://en.cppreference.com/w/cpp/feature_test#Library_features) to know if feature is officially supported (and [`__has_include`](https://en.cppreference.com/w/cpp/preprocessor/include) too)...
